### ISSUE TITLE
Simplify CLI

### ISSUE THREAD

**Issue by aweiteka** [projectatomic/atomicapp#27]
We're running into some issues when running as a container such as how to pass in an optional conf file and how to pass in params like interactive `--ask`.

The `install` behavior is not intuitive. It should be optional in a container context. My understanding is it pulls in remote sources (if there are any) and moves the files to a new directory. I suggest we default to current working directory `${PWD}` and leave things there, adding an `external` directory during install for remote sources. Options would be `--source <src_dir>` and `--destination <dst_dir>`. No strong opinion on option names; whatever makes sense.

In a container context bindmount options are simplified as `-v ${PWD}:/atomicapp`. The `answerfile.conf` will appear there (if present). If `/atomicapp` doesn't have a `Nulecule` file then copy contents of `cp /application-entity/* /atomicapp/ -r`. We won't want to do this in a non-container context.

**Comment by vpavlin** [comment]
Current behvaiour

```
[vpavlin@dhcp-30-155 ~/devel/upstream/atomicapp-run/test-app(master)]
$ ls
[vpavlin@dhcp-30-155 ~/devel/upstream/atomicapp-run/test-app(master)]
$ atomicapp install vpavlin/helloapache
2015-05-08 12:21:39,267 - atomicapp.install - INFO - App name is vpavlin/helloapache, will be populated to None
2015-05-08 12:21:39,267 - atomicapp.params - WARNING - No answers file found.
Trying to pull repository docker.io/vpavlin/helloapache ...
0f330701741d: Download complete 
6941bfcbbfca: Download complete 
41459f052977: Download complete 
fd44297e2ddb: Download complete 
f286a6c39e45: Download complete 
73846768f6ac: Download complete 
d7b655f6e4dc: Download complete 
b2d4efd88915: Download complete 
5bce9d791f66: Download complete 
d0cfb7fc6a83: Download complete 
922a34ba98de: Download complete 
de84a0b71273: Download complete 
3e7e3dcc0576: Download complete 
d27de2504144: Download complete 
064327e7515d: Download complete 
209139e515c1: Download complete 
1971e8e9c2d9: Download complete 
6fa124794b79: Download complete 
d73fcf4d4871: Download complete 
Status: Downloaded newer image for docker.io/vpavlin/helloapache:latest
973c06b1321b2c69d05ee99237cf38a6600cea48cd62f6dfc388529f115eaa76
2015-05-08 12:22:36,186 - atomicapp.utils - INFO - Using temporary directory /tmp/appent-helloapachenJG9kV
helloapache
2015-05-08 12:22:37,040 - atomicapp.install - INFO - Copying app helloapache
2015-05-08 12:22:37,045 - atomicapp.utils - INFO - Artifacts for helloapache-app present for these providers: kubernetes
[vpavlin@dhcp-30-155 ~/devel/upstream/atomicapp-run/test-app(master)]
$ ls
Dockerfile  graph  nulecule
```

`install` also accepts path (which is used in LABELS) and is probably what you want in this issue. I guess we could change the param `--path` to `--destination`..am I missing something?

**Comment by vpavlin** [comment]
To give you a complete example:

```
$ atomicapp install -p dest-dir vpavlin/helloapache
2015-05-08 12:43:57,076 - atomicapp.install - INFO - App name is vpavlin/helloapache, will be populated to dest-dir
2015-05-08 12:43:57,076 - atomicapp.params - WARNING - No answers file found.
Trying to pull repository docker.io/vpavlin/helloapache ...
0f330701741d: Download complete 
6941bfcbbfca: Download complete 
41459f052977: Download complete 
fd44297e2ddb: Download complete 
f286a6c39e45: Download complete 
73846768f6ac: Download complete 
d7b655f6e4dc: Download complete 
b2d4efd88915: Download complete 
5bce9d791f66: Download complete 
d0cfb7fc6a83: Download complete 
922a34ba98de: Download complete 
de84a0b71273: Download complete 
3e7e3dcc0576: Download complete 
d27de2504144: Download complete 
064327e7515d: Download complete 
209139e515c1: Download complete 
1971e8e9c2d9: Download complete 
6fa124794b79: Download complete 
d73fcf4d4871: Download complete 
Status: Image is up to date for docker.io/vpavlin/helloapache:latest
fb650af5720923366c0cc0c1916f64e4706a7670ee3e2c2f949605c55c99d288
2015-05-08 12:44:05,423 - atomicapp.utils - INFO - Using temporary directory /tmp/appent-helloapacheKnsArn
helloapache
2015-05-08 12:44:05,899 - atomicapp.install - INFO - Copying app helloapache
2015-05-08 12:44:05,903 - atomicapp.utils - INFO - Artifacts for helloapache-app present for these providers: kubernetes
[vpavlin@dhcp-30-155 ~/devel/upstream/atomicapp-run(master)]
$ ls dest-dir/
Dockerfile  graph  nulecule
```

And the `src_dir -> dst_dir` case:

```
$ ls src-dir/
Dockerfile  graph  nulecule
[vpavlin@dhcp-30-155 ~/devel/upstream/atomicapp-run(master)]
$ ls dest-dir
ls: nelze přistoupit k dest-dir: Adresář nebo soubor neexistuje
[vpavlin@dhcp-30-155 ~/devel/upstream/atomicapp-run(master)]
$ atomicapp install -p dest-dir src-dir/
2015-05-08 12:46:19,355 - atomicapp.install - INFO - App path is src-dir/, will be populated to dest-dir
2015-05-08 12:46:19,362 - atomicapp.params - WARNING - No answers file found.
2015-05-08 12:46:19,363 - atomicapp.install - INFO - Copying content of directory src-dir/ to dest-dir
2015-05-08 12:46:19,363 - atomicapp.install - INFO - Copying app helloapache-app
2015-05-08 12:46:19,365 - atomicapp.utils - INFO - Artifacts for helloapache-app present for these providers: kubernetes
2015-05-08 12:46:19,365 - atomicapp.install - INFO - Component data exist in dest-dir/graph/helloapache-app, skipping population...
[vpavlin@dhcp-30-155 ~/devel/upstream/atomicapp-run(master)]
$ ls dest-dir/
Dockerfile  graph  nulecule
```

So my suggestion is to change `-p|--path` to `--destination` (I am not sure if we need/want short option or what letter it could be) but I am not sure about adding `--source` option as I think it is not needed. But I am definitely not a UX expert:)

**Comment by aweiteka** [comment]
Here's the end goal from a user experience perspective. Note: this is running from a container perspective.

1. I have a known app I want to deploy. I have an answerfile prepared in my local current working directory.

        $ ls -1
        answers.conf
        $ sudo atomic run my/app

2. Answerfile creation. Two paths:
    1. I have an app I want to deploy but I need help with the answerfile. I want to download the composite answerfile template, add my values and run. But first I want to see what it's going to do.

        $ ls
           # nothing here
        $ sudo atomic install my/app
        $ vi answers.conf
        $ sudo atomic run my/app --dry-run
        running in dry-run mode. Nothing will be deployed.
        kubectl create ...
        kubectl create ...
        ...
        $ sudo atomic run my/app
        deploying to provider...

    1. I have an app I want to deploy but I need help with the answerfile. Since I don't have an answerfile in my current directory I expect to be prompted for the answerfile values before it runs.

        $ ls
           # nothing here
        $ sudo atomic run my/app
        Running in interactive mode...
        db_password? ******
        namespace? my-cool-app
        ...

1. I have an app that I want to add an answerfile to for future unattended installs.

        $ ls
           # nothing here
        $ sudo atomic install my/app
        $ vi answers.conf
        $ sudo docker build -t mynewapp/withanswerfile .
        $ sudo docker push mynewapp/withanswerfile

1. I have an app that was prepped in previous step for unattended installs.

        $ ls
            # nothing here
        $ sudo atomic run mynewapp/withanswerfile

**Comment by vpavlin** [comment]
PRs #34 #35 should solve all above

1. Is solved by #35 specifically
2. For this
 1. `answers.conf.sample` is written out so that user just needs to 

    ```
mv answers.conf.sample answers.conf
vi answers.conf
    ```
 2. As the file `answers.conf.sample` is outputed and thus `answers.conf` does not exist, user is queried for missing values

3. Works, user just needs to add the following line to the Dockerfile

 ```
ADD /answers.conf /application-entity/
 ```

4. I like this and it works if you call `install` and then `run`, but does not work directly with `run` cause we pass `/atomicapp` to the `LABEL RUN` - i.e. we need the data already installed. I think the workaround could be to look at ENVs and specifically `$IMAGE` which is set in our `LABEL RUN` and thus the `run` could simply use the given path as a `--destination` for install.

 ```
bash-4.2# rm -rf *
bash-4.2# atomic install vpavlin/helloapache-conf
docker run --rm -it --privileged -v /run:/run -v ${PWD}:/atomicapp -e IMAGE=vpavlin/helloapache-conf -e NAME=helloapache-conf --name helloapache-conf vpavlin/helloapache-conf -v install --destination /atomicapp /application-entity
2015-05-09 16:02:38,233 - atomicapp.install - INFO - App path is /application-entity, will be populated to /atomicapp
2015-05-09 16:02:38,239 - atomicapp.params - DEBUG - Setting app id to helloapache-app
...
2015-05-09 16:02:38,243 - atomicapp.install - DEBUG - {u'helloapache-app': {u'image': u'fedora/apache'}}
2015-05-09 16:02:38,243 - atomicapp.params - DEBUG - Data given {u'helloapache-app': {u'image': u'fedora/apache'}}
2015-05-09 16:02:38,243 - atomicapp.install - DEBUG - {u'helloapache-app': {u'image': u'fedora/apache'}, 'general': {'provider': 'kubernetes'}}
blah
2015-05-09 16:02:38,243 - atomicapp.params - INFO - Writing answers file template to /atomicapp/answers.conf.sample
bash-4.2# ls
answers.conf  answers.conf.sample  Dockerfile  graph  nulecule
 ```

But I think all parts of this issue are solved by the mentioned PRs

**Comment by vpavlin** [comment]
Ok, the 4. is fixed by last commit:

```
bash-4.2# rm -rf *
bash-4.2# atomicapp run vpavlin/helloapache-conf
2015-05-09 16:29:36,726 - atomicapp.install - INFO - App name is vpavlin/helloapache-conf, will be populated to /var/home/centos/atomicapp/test-hello
3880d39ef06229b0cfc3d49f328fd9d9eed0a25cae9d5ab7fc7992d82daa359a
2015-05-09 16:29:37,700 - atomicapp.utils - INFO - Using temporary directory /tmp/appent-helloapache-confv7gk3K
helloapache-conf
2015-05-09 16:29:38,298 - atomicapp.install - INFO - Copying app helloapache-conf
2015-05-09 16:29:38,303 - atomicapp.utils - INFO - Artifacts for helloapache-app present for these providers: kubernetes
2015-05-09 16:29:38,303 - atomicapp.install - INFO - Installing dependencies for helloapache-app
blah
2015-05-09 16:29:38,303 - atomicapp.params - INFO - Writing answers file template to /var/home/centos/atomicapp/test-hello/answers.conf.sample
/var/home/centos/atomicapp/atomicapp/providers/docker.py:1: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  from atomicapp.plugin import Provider
/var/home/centos/atomicapp/atomicapp/providers/docker.py:2: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  import os, subprocess
/var/home/centos/atomicapp/atomicapp/providers/docker.py:4: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  import logging
/var/home/centos/atomicapp/atomicapp/providers/openshift.py:1: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  from atomicapp.plugin import Provider
/var/home/centos/atomicapp/atomicapp/providers/openshift.py:3: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  from collections import OrderedDict
/var/home/centos/atomicapp/atomicapp/providers/openshift.py:5: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  import os, subprocess
/var/home/centos/atomicapp/atomicapp/providers/openshift.py:6: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  import anymarkup
/var/home/centos/atomicapp/atomicapp/providers/openshift.py:7: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  import logging
/var/home/centos/atomicapp/atomicapp/providers/kubernetes.py:1: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  from atomicapp.plugin import Provider
/var/home/centos/atomicapp/atomicapp/providers/kubernetes.py:3: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  from collections import OrderedDict
/var/home/centos/atomicapp/atomicapp/providers/kubernetes.py:4: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  import os, json, subprocess
/var/home/centos/atomicapp/atomicapp/providers/kubernetes.py:6: RuntimeWarning: Parent module 'atomicapp.providers' not found while handling absolute import
  import logging
2015-05-09 16:29:38,312 - atomicapp.utils - INFO - Artifacts for helloapache-app present for these providers: kubernetes
2015-05-09 16:29:38,312 - atomicapp.utils - INFO - Using temporary directory /tmp/appent-helloapache-appWdDHlw
2015-05-09 16:29:38,313 - atomicapp.run - INFO - Using provider kubernetes for component helloapache-app
Calling: kubectl create -f /tmp/appent-helloapache-appWdDHlw/helloapache-app/graph/k8s/hello-apache-pod.json
pods/fedoraapache
```
